app/main.py

```diff
@@ -1,52 +1,40 @@
-import json
-import xml.etree.ElementTree as ET
+from app.abstract_classes import IBookDisplay, IBookPrint, IBookSerializer
+from app.book_display import ConsoleBookDisplay, ReverseBookDisplay
+from app.book_model import Book
+from app.book_print import ConsoleBookPrint, ReverseBookPrint
+from app.serializers import JSONBookSerializer, XMLBookSerializer
 
 
-class Book:
-    def __init__(self, title: str, content: str):
-        self.title = title
-        self.content = content
-
-    def display(self, display_type: str) -> None:
-        if display_type == "console":
-            print(self.content)
-        elif display_type == "reverse":
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown display type: {display_type}")
-
-    def print_book(self, print_type: str) -> None:
-        if print_type == "console":
-            print(f"Printing the book: {self.title}...")
-            print(self.content)
-        elif print_type == "reverse":
-            print(f"Printing the book in reverse: {self.title}...")
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown print type: {print_type}")
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
-        else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
-
-
-def main(book: Book, commands: list[tuple[str, str]]) -> None | str:
+def main(book: Book, commands: list[tuple[str, str]]) -> str | None:
     for cmd, method_type in commands:
         if cmd == "display":
-            book.display(method_type)
```

Comment:
You shouldnt perform those check here. It need to be incapsulated in a class which you should use here instead
There same is for print and serialize commands